dace/symbolic.py

```diff
@@ -854,3 +854,43 @@ def persistent_load(self, pid):
             return _sunpickle(value)
         else:
             raise pickle.UnpicklingError("unsupported persistent object")
+    
+
+def equalize_symbol(sym: sympy.Expr) -> sympy.Expr:
+    """ If a symbol or symbolic expressions has multiple symbols with the same
+        name, it substitutes them with the last symbol (as they appear in
+        s.free_symbols).
+    """
+    symdict = {s.name: s for s in sym.free_symbols}
+    repldict = {s: symdict[s.name] for s in sym.free_symbols}
+    return sym.subs(repldict)
+
+
+def equalize_symbols(a: sympy.Expr, b: sympy.Expr) -> Tuple[sympy.Expr,
+                                                            sympy.Expr]:
+    """ If the 2 input expressions use different symbols but with the same name,
+        it substitutes the symbols of the second expressions with those of the
+        first expression.
+    """
+    a = equalize_symbol(a)
+    b = equalize_symbol(b)
+    a_syms = {s.name: s for s in a.free_symbols}
+    b_syms = {s.name: s for s in b.free_symbols}
+    common_names = set(a_syms.keys()).intersection(set(b_syms.keys()))
+    if common_names:
+        repldict = dict()
+        for name in common_names:
+            repldict[b_syms[name]] = a_syms[name]
+        b = b.subs(repldict)
+    return a, b
+
+
+def inequal_symbols(a: Union[sympy.Expr, Any],
+                    b: Union[sympy.Expr, Any]) -> bool:
+    """ Compares 2 symbolic expressions and returns True if they are not equal.
```

Comment:
```suggestion
    """
    Compares 2 symbolic expressions and returns True if they are not equal.
```

Reply:
I don't understand the suggestion? Is it to move the text one line below? Why?

Reply:
just a matter of consistent style.
```
"""
doc
:param x:
:param y:
:return:
:note:
:raises z:
"""
```

Reply:
Done